### ISSUE TITLE
Add support for Proc/lambda on column_types

### ISSUE THREAD

**Issue by westonganger** [westonganger/spreadsheet_architect#50]
It would be good to support Proc/lambda for column types

This could be supported as 

```ruby
.to_xlsx(
  column_types: [
    :string, 
    ->(cell_val){ cell_val.start_with?("http") ? :hyperlink : :string }, 
    :float,
  ]
)
```

We dont need to do anything additional for `spreadsheet_columns` because its already instance evaluated so you can already do things like

```ruby
.to_xlsx(
  spreadsheet_columns: ->(){
    [
      "Foo", :foo, :string
      "Bar", :bar, (bar.start_with?("http") ? :hyperlink : :string)
      "Baz", :baz, :float
    ]
  }
)
```

**Comment by westonganger** [comment]
This is being implemented in #51 if you want to try this out

**Comment by westonganger** [comment]
This is now implemented in master.
